src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -0,0 +1,269 @@
+#include "application_manager/resumption/resume_ctrl_impl.h"
+#include "application_manager/mock_application_manager_settings.h"
+#include "hmi_message_handler/mock_hmi_message_handler.h"
+#include "connection_handler/mock_connection_handler.h"
+#include "protocol_handler/mock_session_observer.h"
+#include "policy/usage_statistics/mock_statistics_manager.h"
+#include "application_manager/test/include/application_manager/mock_message_helper.h"
+
+namespace test {
+namespace components {
+namespace application_manager_test {
+
+namespace am = application_manager;
+namespace policy_test = test::components::policy_handler_test;
+namespace con_test = connection_handler_test;
+
+using testing::_;
+using ::testing::Mock;
+using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::NiceMock;
+using ::testing::SetArgReferee;
+using ::testing::SetArgPointee;
+using ::testing::Truly;
+using ::testing::AtLeast;
+using ::testing::DoAll;
+using usage_statistics_test::MockStatisticsManager;
+using namespace application_manager;
+
+namespace {
+const std::string kHmiCapabilitiesFile("hmi_capabilities.json");
+const uint32_t kThreadPoolSize = 1;
+const bool kHmiLaunch = false;
+const uint32_t kStopStreamingTimeout = 10000;
+const uint32_t kWaitFreeQueue = 2000;
+const std::string kDirectoryName = "./test_storage";
+const uint32_t kTimeout = 10000u;
+const std::string kFolderName = "app";
+// const std::string kTestPath =
+//    file_system::ConcatPath(kDirectoryName, kFolderName);
+// const std::string kTestFileName =
+//    file_system::ConcatPath(kTestPath, "test_file2.bmp");
+const std::string kMacAddr_("kMacAddr_ess");
+
+bool is_called_ = false;
+uint32_t hmi_app_id_ = 0u;
+sync_primitives::Lock state_lock_;
+sync_primitives::ConditionalVariable state_condition_;
+}  // namespace
+
+class ApplicationManagerImplTest : public ::testing::Test {
+ public:
+  ApplicationManagerImplTest()
+      : mock_message_helper_(
+            application_manager::MockMessageHelper::message_helper_mock()) {}
+
+ protected:
+  void SetUp() OVERRIDE {
+    app_id_ = 10;
+
+    Mock::VerifyAndClearExpectations(&mock_message_helper_);
+
+    CreateAppManager();
+
+    //    ON_CALL(mock_session_observer_, GetDataOnDeviceID(_, _, _, _, _))
+    //        .WillByDefault(DoAll(SetArgPointee<3u>(kMacAddr_), Return(0)));
+    ON_CALL(mock_connection_handler_, GetDataOnSessionKey(_, _, _, _))
+        .WillByDefault(DoAll(SetArgPointee<3u>(app_id_), Return(0)));
+    ON_CALL(mock_connection_handler_, get_session_observer())
+        .WillByDefault(ReturnRef(mock_session_observer_));
+
+    mock_storage_ =
+        ::utils::MakeShared<NiceMock<resumption_test::MockResumptionData> >(
+            app_mngr_);
+    // app_mock = utils::MakeShared<NiceMock<MockApplication> >();
+    // res_ctrl = utils::MakeShared<ResumeCtrlImpl>(app_mngr_);
+    // res_ctrl->set_resumption_storage(mock_storage);
+
+    logger::create_log_message_loop_thread();
```

Comment:
@pold500 are you sure that you should create logger loop thread there? At least I suppose that it should be create before any other activities